phenotrex/transforms/annotation.py

```diff
@@ -0,0 +1,115 @@
+import os
+from typing import List
+from pathlib import Path
+from pkg_resources import resource_filename
+from tempfile import NamedTemporaryFile
+from subprocess import check_call, DEVNULL
+from concurrent.futures import ProcessPoolExecutor
+
+import numpy as np
+import torch
+from Bio.SeqIO import SeqRecord, parse
+from deepnog.deepnog import load_nn, predict, set_device, create_df
+from deepnog.dataset import ProteinDataset
+from tqdm.auto import tqdm
+
+from phenotrex.io.flat import load_fasta_file
+from phenotrex.structure.records import GenotypeRecord
+
+PRODIGAL_BIN_PATH = resource_filename('phenotrex', 'bin/prodigal')
+DEEPNOG_WEIGHTS_PATH = resource_filename('deepnog', 'parameters/eggNOG5/2/deepencoding.pth')
+DEEPNOG_ARCH = 'deepencoding'
+
+
+class PreloadedProteinDataset(ProteinDataset):
+    """Hack ProteinDataset to load from list directly."""
+    def __init__(self, protein_list: List[SeqRecord]):
+        try:
+            super().__init__(file='')
+        except ValueError:
+            pass  # >:D
+        self.iter = (x for x in protein_list)
+
+
+def fastas_to_grs(fasta_files: List[str], verb: bool = False,
+                 n_threads: int = None) -> List[GenotypeRecord]:
+    """
+    Perform GenotypeRecord calculation for a list of FASTA files. Apply process-based parallelism
+    since gene annotation scales well with cores.
+
+    :param fasta_files: a list of DNA and/or protein FASTA files to be converted into GenotypeRecords.
+    :param verb: Whether to display progress of annotation with tqdm.
+    :param n_threads: Number of threads to run in parallel. Default, use up to all available CPU cores.
+    :returns: A list of GenotypeRecords corresponding with supplied FASTA files.
+    """
+    n_threads = np.min([os.cpu_count(), n_threads]) if n_threads is not None else os.cpu_count()
```

Comment:
```suggestion
    n_threads = min(os.cpu_count(), n_threads) if n_threads is not None else os.cpu_count()
```
No need for numpy. Line is long enough already.

Reply:
def helloworld():
    import numpy as np
    import sys
    np.savetxt(sys.stdout.buffer, np.array(['hello, world!', ]), fmt='%s')